src/buffer/out/textBuffer.cpp

```diff
@@ -944,6 +944,115 @@ Microsoft::Console::Render::IRenderTarget& TextBuffer::GetRenderTarget() noexcep
     return _renderTarget;
 }
 
+// Method Description:
+// - Get the COORD for the beginning of the word you are on
+// Arguments:
+// - target - a COORD on the word you are currently on
+// - wordDelimiters - what characters are we considering for the separation of words
+// - includeCharacterRun - include the character run located at the beginning of the word
+// Return Value:
+// - The COORD for the first character on the "word"  (inclusive)
+const COORD TextBuffer::GetWordStart(const COORD target, const std::wstring wordDelimiters, bool includeCharacterRun) const
+{
+    COORD result = target;
+
+    // can't expand left
+    if (target.X == GetSize().Left())
+    {
+        return result;
+    }
+
+    auto bufferIterator = GetTextDataAt(result);
+    const auto startedOnDelimiter = _GetDelimiterClass(*bufferIterator, wordDelimiters);
+    while (result.X > GetSize().Left() && (_GetDelimiterClass(*bufferIterator, wordDelimiters) == startedOnDelimiter))
+    {
+        GetSize().DecrementInBounds(result);

```

Comment:
We should probably cache this `GetSize()` result once near the top of this function - it's not going to change during this function, and it's not like it's returning a reference or something - I think it's constructing a `Viewport` each call